File: src/my_cpp_pkg/src/node.cpp
#include <rclcpp/rclcpp.hpp>
#include <chrono>

int counter = 0;

void timer_callback()
{
    counter++;

    RCLCPP_INFO(
        rclcpp::get_logger("my_node"),
        "Timer callback triggered! Count: %d",
        counter
    );
}

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<rclcpp::Node>("my_node");
    RCLCPP_INFO(node->get_logger(), "Hello, ROS 2!");
    auto timer = node->create_wall_timer(
        std::chrono::seconds(1),
        timer_callback
    );
    rclcpp::spin(node);
   
    // Destroy timer
    timer->cancel();
    timer.reset();
   
    // Destroy node
    node.reset();
    rclcpp::shutdown();

    return 0;
}

File: src/my_py_pkg/my_py_pkg/node.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node

class MyNode(Node):
    def __init__(self):
        super().__init__("my_node")
        self.declare_parameter("counter", 0)
        self.declare_parameter("frequency", 1.0)  
        self.counter_ = self.get_parameter("counter").value
        self.frequency_ = self.get_parameter("frequency").value 

        self.create_timer(1.0 / self.frequency_, self.timer_callback)
        self.get_logger().info("Hello, ROS 2!")

    def timer_callback(self):
        self.counter_ += 1
        self.get_logger().info(
            f"Timer callback executed {self.counter_} times.")

def main(args=None):
    rclpy.init(args=args)
    node = MyNode()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()

if __name__ == "__main__":
    main()